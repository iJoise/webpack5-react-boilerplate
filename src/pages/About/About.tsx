import React from "react";
import s from "./style/About.module.scss";
import logo from "../../assets/icons/logo.svg";
import { ReactComponent as Logo } from "../../assets/icons/logo2.svg";
import { useGenerateColor } from "../../hooks/useGenerateColor";
import { Button } from "../../shared/Button";

export const About = () => {
  const { randomColor, generateColor } = useGenerateColor();

  return (
    <div className={s.about}>
      <h1 className={s.about__title}>React Boilerplate</h1>
      <h2 className={s.about__subtitle}>Happy coding!</h2>
      <div className={s.about__body}>
        <div className={s.about__column}>
          <pre>{"<img src={logo} />"}</pre>
          <img src={logo} className={s.about__image} alt="logo" />
        </div>

        <div className={s.about__column}>
          <pre>ReactComponent</pre>
          <Logo
            data-testid="react-svg"
            width={200}
            height={200}
            fill={randomColor}
            className={s.about__image}
          />
          <Button className={s.about__btn} rounded type="button" onClick={generateColor}>
            Set color
          </Button>
        </div>

        <div className={s.about__column}>
          <pre>background-image</pre>
          <div className={s.about__background} />
        </div>
      </div>
    </div>
  );
};
