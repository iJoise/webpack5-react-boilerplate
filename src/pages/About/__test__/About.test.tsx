import React from "react";
import { act, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { About } from "../About";

describe("About Page", () => {
  it("About Page should be render", () => {
    const { container } = render(<About />);
    expect(screen.getByText(/react boilerplate/i)).toBeInTheDocument();
    expect(screen.getByText(/happy coding!/i)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /set color/i })).toBeInTheDocument();
    expect(screen.getByTestId("react-svg")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("SVG color should change correctly", () => {
    render(<About />);

    const reactIcon = screen.getByTestId("react-svg");
    const currentFill = reactIcon.getAttribute("fill");
    const button = screen.getByRole("button", { name: /set color/i });

    act(() => {
      fireEvent.click(button);
    });

    const newFill = screen.getByTestId("react-svg").getAttribute("fill");
    expect(newFill).not.toEqual(currentFill);
  });
});
