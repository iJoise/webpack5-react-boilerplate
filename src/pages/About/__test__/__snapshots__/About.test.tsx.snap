// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page About Page should be render 1`] = `
<div>
  <div
    class="about"
  >
    <h1
      class="about__title"
    >
      React Boilerplate
    </h1>
    <h2
      class="about__subtitle"
    >
      Happy coding!
    </h2>
    <div
      class="about__body"
    >
      <div
        class="about__column"
      >
        <pre>
          &lt;img src={logo} /&gt;
        </pre>
        <img
          alt="logo"
          class="about__image"
          src="[object Object]"
        />
      </div>
      <div
        class="about__column"
      >
        <pre>
          ReactComponent
        </pre>
        <svg
          class="about__image"
          data-testid="react-svg"
          fill="rgb(129, 178, 154)"
          height="200"
          width="200"
        />
        <button
          class="about__btn button button__rounded"
          type="button"
        >
          Set color
        </button>
      </div>
      <div
        class="about__column"
      >
        <pre>
          background-image
        </pre>
        <div
          class="about__background"
        />
      </div>
    </div>
  </div>
</div>
`;
